test/widgets/lightbox_test.dart

```diff
@@ -558,4 +601,51 @@ void main() {
       check(platform.position).equals(position);
     });
   });
-}
+
+  group('LightboxHero', () {
+    testWidgets('no hero animation occurs between different message list pages for same image', (tester) async {
+      final channel = eg.stream(streamId: eg.defaultStreamMessageStreamId);
+      final message = eg.streamMessage(stream: channel,
+        contentMarkdown: ContentExample.spoilerHeaderHasImage.html, topic: 'test topic');
+      await setupMessageListPage(tester, narrow: const CombinedFeedNarrow(),
+        streams: [channel], messages: [message]);
+
+      connection.prepare(json:
+        eg.newestGetMessagesResult(foundOldest: true, messages: [message]).toJson());
+
+      final imageFinder = find.byType(RealmContentNetworkImage).first;
+      final initialImageRect = tester.getRect(imageFinder);
+
+      await tester.tap(find.descendant(
+        of: find.byType(StreamMessageRecipientHeader),
+        matching: find.text('test topic')));
+      await tester.pump();
+      await tester.pump(const Duration(milliseconds: 150));
+
+      final imageInTransition = tester.getRect(imageFinder);
+      check(imageInTransition.top).equals(initialImageRect.top);
+      check(imageInTransition.left).equals(initialImageRect.left);
+      await tester.pumpAndSettle();
+    });
+
+    testWidgets('hero animation occurs when opening lightbox from message list', (tester) async {
+      final channel = eg.stream(streamId: eg.defaultStreamMessageStreamId);
+      final message = eg.streamMessage(stream: channel,
+        contentMarkdown: ContentExample.spoilerHeaderHasImage.html);
+      await setupMessageListPage(tester, narrow: const CombinedFeedNarrow(),
+        streams: [channel], messages: [message]);
+
+      final imageFinder = find.byType(RealmContentNetworkImage);
+      final initialImageRect = tester.getRect(imageFinder);
+
+      await tester.tap(imageFinder);
+      await tester.pump();
+      await tester.pump(const Duration(milliseconds: 150));
+
+      final imageInTransition = tester.getRect(imageFinder);
+      check(imageInTransition.top != initialImageRect.top).isTrue();
```

Comment:
nit: utilize checks API when possible:
```suggestion
      check(imageInTransition.top).not((it) => it.equals(initialImageRect.top));
```
same below